### PR TITLE
Quizzes

### PR DESCRIPTION
**Pull request by msdeshmukh009**
### Added quizzes
orion-quiz

### CODE REVIEW

File: src/components/category-card/CategoryCard.tsx [hunk 1]
```diff
@@ -19,7 +19,10 @@ const CategoryCard = ({ category }: CategoryCardProps) => {
       </div>
       {categoryStatus === "available" ? (
         <div className="card-cta-vertical">
-          <Link to="explore-quiz" className="btn btn-primary block-btn text-center">
+          <Link
+            to={`explore-quiz/${categoryName.toLocaleLowerCase().replaceAll(" ", "-")}`}
```
Comment (Chakravarthi0) [inline src/components/category-card/CategoryCard.tsx:23]: suggestion: you can extract it as a function

File: src/components/quiz-card/QuizCard.tsx [hunk 2]
```diff
@@ -1,23 +1,42 @@
...
+    }
+  };
+
   return (
     <div className="card vertical-card max-w-xs shadow">
       <div className="card-image-container">
-        <img className="responsive-img rounded-top-corner-img" src={quizImage} alt={quizName} />
+        <img
+          width="300"
+          height="168"
```
Comment (Chakravarthi0) [inline src/components/quiz-card/QuizCard.tsx:25]: You can move this to css

File: src/components/result-card/ResultCard.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  question,
+  options,
+  questionIndex,
+  correctOption,
+}: {
+  question: string;
+  options: string[];
+  questionIndex: number;
+  correctOption: number;
+}) => {
```
Comment (Chakravarthi0) [inline src/components/result-card/ResultCard.tsx:13]: suggestion: similar to context types, create a component types folder and have all the component types there.

File: src/utils/firebaseHelpers.ts [hunk 4]
```diff
@@ -41,4 +41,30 @@ const getUserDocument = async (uid: string) => {
...
+};
+
+const getQuiz = async (quizId: string) => {
+  const quizRef = collection(db, `quizzes/${quizId}/questions`);
+  const res = await getDocs(quizRef);
+  const quiz: DocumentData | undefined = res.docs.map(ele => {
+    return { ...ele.data(), id: ele.id };
+  });
+  return quiz;
+};
```
Comment (Chakravarthi0) [inline src/utils/firebaseHelpers.ts]: you can add try catch blocks here

**Review by Chakravarthi0 — APPROVED** [review]
app is working fine.
apard from the esuggestions, everything looks good.

File: src/components/category-card/CategoryCard.tsx (hunk 1 above)
Comment (msdeshmukh009) [reply src/components/category-card/CategoryCard.tsx:23]: sure i will

File: src/components/result-card/ResultCard.tsx (hunk 3 above)
Comment (msdeshmukh009) [reply src/components/result-card/ResultCard.tsx:13]: ok will refactor in the next pr

File: src/utils/firebaseHelpers.ts (hunk 4 above)
Comment (msdeshmukh009) [reply src/utils/firebaseHelpers.ts]: ok adding now.

File: src/components/quiz-card/QuizCard.tsx [hunk 5]
```diff
@@ -1,23 +1,42 @@
...
+  const { id, quizName, quizDescription, quizStatus, quizImage } = quiz;
+  const { getQuestions } = useGame();
+  const navigate = useNavigate();
+
+  const handlePlayNow = async () => {
+    try {
+      await getQuestions(id, quizName);
+      navigate("/rules");
+    } catch (err) {
+      console.log(err);
```
Comment (manojasarsa) [inline src/components/quiz-card/QuizCard.tsx:16]: You can use `console.error` here

**Review by manojasarsa — APPROVED** [review]
The code is structured very well.
App is working as expected.
UI is also good.
GTG!

File: src/context/QuizContext.tsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+const QuizContext = createContext({} as QuizContextType);
+
+const QuizProvider = ({ children }: ReactChildren) => {
+  const [quizState, quizDisPatch] = useReducer(quizReducer, quizInitialState);
+
+  useEffect(() => {
+    (async () => {
+      try {
+        quizDisPatch({ type: "INITIALIZE" });
+        const response = await getCategories();
```
Comment (poornima-alugubelly) [inline src/context/QuizContext.tsx:16]: Can we use keep all these constants in a constants file and reuse them instead writing a string here?

File: src/pages/home/Home.tsx [hunk 7]
```diff
@@ -61,9 +44,18 @@ const Home = () => {
...
-            })}
+            {isLoading && <Loading />}
+            {categoryData?.map(
+              (category: {
+                id: string;
+                categoryDescription: string;
+                categoryImage: string;
+                categoryName: string;
+                categoryStatus: string;
+              }) => {
```
Comment (poornima-alugubelly) [inline src/pages/home/Home.tsx:55]: As suggested by Charan you could extract these types as well

**Review by poornima-alugubelly — APPROVED** [review]
Website is working well in both desktops and mobile modes , good work 👍

File: src/components/quiz-card/QuizCard.tsx (hunk 5 above)
Comment (msdeshmukh009) [reply src/components/quiz-card/QuizCard.tsx:16]: sure i will add proper error handling here in next pr

File: src/context/QuizContext.tsx (hunk 6 above)
Comment (msdeshmukh009) [reply src/context/QuizContext.tsx:16]: As I have strictly typed every action-type typescript will warn me if I use any invalid action type that's why I did not convert these strings to variables.

File: src/pages/home/Home.tsx (hunk 7 above)
Comment (msdeshmukh009) [reply src/pages/home/Home.tsx:55]: Sure I will refactor in the next pr. Thanks for the review!!

**Pull request merged**